src/main/java/frc/robot/subsystems/drivetrain/commands/HolonomicDrive.java

```diff
@@ -0,0 +1,53 @@
+package frc.robot.subsystems.drivetrain.commands;
+
+import edu.wpi.first.wpilibj.GenericHID;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.*;
+import frc.robot.subsystems.drivetrain.SwerveDrive;
+import org.techfire225.webapp.FireLog;
+
+public class HolonomicDrive extends CommandBase {
+
+    private SwerveDrive swerveDrive;
+
+    public HolonomicDrive(SwerveDrive swerveDrive) {
+        this.swerveDrive = swerveDrive;
+        addRequirements(swerveDrive);
+    }
+
+    @Override
+    public void initialize() {
+        Robot.gyro.reset();
+
+    }
+
+    @Override
+    public void execute() {
+        double forward = Utils.joystickDeadband(-OI.xbox.getY(GenericHID.Hand.kRight));
+        double strafe = Utils.joystickDeadband(-OI.xbox.getX(GenericHID.Hand.kRight));
+        double rotation = Utils.joystickDeadband(-OI.xbox.getX(GenericHID.Hand.kLeft));
+        if (forward != 0 || strafe != 0 || rotation != 0) {
+            swerveDrive.holonomicDrive(forward, strafe, rotation);
+        } else {
+            for (int i = 0; i < 4; i++) {
+                swerveDrive.swerveModules[i].setSpeed(0);
+                swerveDrive.swerveModules[i].setAngle(swerveDrive.swerveModules[i].getAngle());
+            }
```

Comment:
why don't you use the `stop()` function instead?

Reply:
We want the wheels to keep their current position

Reply:
and what is the difference? inside the `stop` function you stop every `driveMotor` and set the power to apply by the `angleMotor` to 0.

Reply:
here the angle motor continues to apply power to make sure the will stays on the same angle

Reply:
Extract this "locking" logic into a method of `SwerveModule`, you use this logic in more than one place.